src/somef/process_repository.py

```diff
@@ -20,14 +33,14 @@ def rate_limit_get(*args, backoff_rate=2, initial_backoff=1, **kwargs):
     date = ""
     while rate_limited:
         response = requests.get(*args, **kwargs)
-        data = response
-        date = data.headers["date"]
-        rate_limit_remaining = data.headers["x-ratelimit-remaining"]
-        epochtime = int(data.headers["x-ratelimit-reset"])
-        date_reset = datetime.fromtimestamp(epochtime)
-        logging.info("Remaining GitHub API requests: " + rate_limit_remaining + " ### Next rate limit reset at: " + str(
+        date = response.headers["Date"]
+        # Show rate limit information if available
+        if "X-RateLimit-Remaining" in response.headers:
+            rate_limit_remaining = response.headers["X-RateLimit-Remaining"]
+            epochtime = int(response.headers["X-RateLimit-Reset"])
+            date_reset = datetime.fromtimestamp(epochtime)
+            logging.info("Remaining GitHub API requests: " + rate_limit_remaining + " ### Next rate limit reset at: " + str(
             date_reset))
-        response = response.json()
```

Comment:
why removing the .json line?

Reply:
In this version of `process_repository.py`, all calls to the GitHub REST API go through `rate_limit_get`. `rate_limit_get` sets up the authorization token, so this way the whole scraping process works on a private repo. Some calls to the GitHub REST API don't get JSON in return, causing `rate_limit_get` to fail. To fix this issue, `rate_limit_get` now returns a raw response, which is decoded to JSON where needed. For example, `response = response.json()` is replace by [this line](https://github.com/KnowledgeCaptureAndDiscovery/somef/pull/592/commits/4fc8fb00bf41c030f1bd67b75b66db47e4b59344#diff-b9f241c1aaafc1a5cf1d638edd19b7af148e6e835ab65a9579f4a7bd9f209178R366) in `load_online_repository_metadata`

Reply:
ok!

Reply:
It turns out the case doesn't matter, because `requests.headers` is of type `requests.structures.CaseInsensitiveDict` :)